src/views/viewer/DiscoverView/DiscoverView.tsx

```diff
@@ -25,22 +26,28 @@ export const DiscoverView: React.FC = () => {
   }))
 
   const categoriesFeaturedVideos = useCategoriesFeaturedVideos()
-  const featuredVideoCategoryCardsData = categories
-    ?.map((category) => {
-      const video =
-        categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
-        categoriesFeaturedVideos?.[category.id]?.[0] ??
-        undefined
-      return video === undefined
-        ? undefined
-        : {
-            videoTitle: video?.title ?? '',
-            videoUrl: video?.videoCutUrl ?? '',
-            ...mappedVideoCategories?.find((cat) => cat.id === category.id),
-          }
-    })
-    .filter((cat) => !!cat) ?? [null, null, null]
+  const featuredVideoCategoryCardsData = React.useMemo(() => {
+    const _featuredVideoCategoryCardsData = sampleSize(
+      categories
+        ?.map((category) => {
+          const video =
+            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
+            categoriesFeaturedVideos?.[category.id]?.[0] ??
+            undefined
```

Comment:
```suggestion
          const video =
            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
            categoriesFeaturedVideos?.[category.id]?.[0]
```
Could be wrong, but I don't think we need this `?? undefined` here.

Reply:
Actually, we can just do: 
```tsx
const video = categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl)
```
Right🤔 ?

Reply:
yes this works